test/integration/kubernetes_deploy_test.rb

```diff
@@ -1006,4 +1007,32 @@ def test_raise_on_yaml_missing_kind
       "  datapoint: value1"
     ], in_order: true)
   end
+
+  def test_crd_can_be_successful
+    assert_deploy_success(deploy_fixtures("hello-cloud", subset: ["crd.yml"]))
+    assert_logs_match_all([
+      "Phase 1: Initializing deploy",
+      "Detected non-namespaced resouce which will never be pruned:",
+      " - CustomResourceDefinition/mails.stable.example.io",
+      "Phase 2: Checking initial resource statuses",
+      "Deploying CustomResourceDefinition/mails.stable.example.io (timeout: 30s)",
+      %r{CustomResourceDefinition/mails.stable.example.io\s+Succeeded}
+    ])
+  end
+
+  def test_crd_can_fail
+    assert_deploy_success(deploy_fixtures("hello-cloud", subset: ["crd.yml"]))
```

Comment:
Is this first deploy necessary in this case? Won't mismatched names be rejected even on the first registration?

Reply:
We're actually creating a conflict between the `listKind` field in the fixture vs. the slightly modified fixture. A mismatch between the name & plural is caught at validation time and isn't really what we want to test.